File: Algorithms/Python/102. Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if not root: return []
        res = []
        stack = [[root]]

        while stack:
            nodes = stack.pop()
            nums = []
            level = []

            for node in nodes:
                nums.append(node.val)
                
                if node.left:
                    level.append(node.left)
                if node.right:
                    level.append(node.right)
            if level:      
                stack.append(level)
                    
            res.append(nums)
            
        return res

            

File: Algorithms/Java/102. Binary Tree Level Order Traversal.java
import java.util.Queue;

import javax.swing.tree.TreeNode;

/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if (root == null) return res;
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);

        while (!queue.isEmpty()) {
            int s = queue.length;
            List<Integer> currentLevel = new ArrayList<>();
            
            for (int i=0; i<s; i++){
                TreeNode current = queue.remove();
                currentLevel.add(current.val);
                if (current.left != null){
                    queue.add(current.left);
                }
                if (current.right != null){
                    queue.add(current.right);
                }
            }
            res.add(currentLevel);
        }        
        return res;
    }
}
